superset-frontend/src/SqlLab/components/ShareSqlLabQuery.tsx

```diff
@@ -74,28 +69,34 @@ class ShareSqlLabQuery extends React.Component {
       })
       .catch(response => {
         getClientErrorObject(response).then(() => {
-          this.props.addDangerToast(t('There was an error with your request'));
+          addDangerToast(t('There was an error with your request'));
         });
       });
-  }
+  };
 
-  getCopyUrlForSavedQuery(callback) {
+  const getCopyUrlForSavedQuery = (callback: Function) => {
     let savedQueryToastContent;
 
-    if (this.props.queryEditor.remoteId) {
+    if (queryEditor.remoteId) {
```

Comment:
do you know if this would be null or undefined? Whichever it is, let's add it to the interface.

Reply:
just to clarify, you mean the remoteId

Reply:
Looking through SqlEditor, it seems like database and latestQuery have a defaultProp of null. Which makes me think that this would be null type as well.

Reply:
Ok, I changed it to null and tested it and the intended functionality is there.